src/Event/Value/Test/TestMethodBuilder.php

```diff
@@ -70,9 +70,11 @@ private static function dataFor(TestCase $testCase): TestDataCollection
                 $dataSetName = (int) $dataSetName;
             }
 
+            $providedData = $testCase->providedData();
+
             $testData[] = DataFromDataProvider::from(
                 $dataSetName,
-                $exporter->export($testCase->providedData()),
+                $exporter->shortenedRecursiveExport($providedData),
                 $testCase->dataSetAsStringWithData(),
```

Comment:
In our usecase, `$testCase->dataSetAsStringWithData()` consumes 99% of a runtime as big data provider arrays are dumper without limit. So I am not sure if this PR itself it really helpful...

The should be some test showing what data are problematic and how this helps.

Reply:
Provide a reproducer for your problem, so we can an verify the fix or build a different one

Reply:
Here is the slow provider testcode - https://github.com/mvorisek/sql-formatter/blob/de3d8d9808f130f7a8734f9e93ec3e216c77003f/tests/TokenizerTest.php#L90-L149.

The only dependency is this simple class - https://github.com/mvorisek/sql-formatter/blob/de3d8d9808f130f7a8734f9e93ec3e216c77003f/src/Token.php.

![image](https://github.com/sebastianbergmann/phpunit/assets/2228672/8165324a-e860-4177-bcb5-b959a6db81f1)

Reply:
I tried reproducing what you described, but I can't see in profiles `dataSetAsStringWithData` taking a relevant share of resources.

please provide a full reproducer.

Reply:
Here is a repro repo: https://github.com/mvorisek/sql-formatter/tree/repro_phpunit_5774_discussion

Repro command: composer update, run phpunit

The phpunit will output something like:
```
...
 Detected 2 tests where the duration exceeded the maximum duration.

1. ...
2. 3.249 (0.500) Doctrine\SqlFormatter\Tests\TokenizerTest::testTokenize#long concat

Time: 00:12.447, Memory: 397.35 MB

OK (283 tests, 294 assertions)
```

/wo dataprovider: https://github.com/doctrine/sql-formatter/actions/runs/9473730952/job/26101927778#step:7:25
/w dataprovider: https://github.com/mvorisek/sql-formatter/actions/runs/9536908812/job/26284385374#step:7:25

As the slowdown is non trivial, you should be able to measure/profile it easily, especially with `--filter 'testTokenize#long concat'` phpunit option.

Reply:
so I can confirm, that this PR does not help for your case.

I wonder why you reported your problem here, as this PR solves a different problem (see PR description).
I think what you are reporting here is not related to this PR and should be reported as a separate issue...?

Reply:
fixed in https://github.com/sebastianbergmann/exporter/pull/59